pow/src/main/java/tech/pegasys/teku/pow/BlockBasedEth1HeadTracker.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.pow;
+
+import static tech.pegasys.teku.util.config.Constants.ETH1_FOLLOW_DISTANCE;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.protocol.core.methods.response.EthBlock.Block;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.subscribers.Subscribers;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.util.config.Constants;
+
+public class BlockBasedEth1HeadTracker implements Eth1HeadTracker {
+  private static final Logger LOG = LogManager.getLogger();
+  private final AtomicBoolean running = new AtomicBoolean(false);
+  private final AsyncRunner asyncRunner;
+  private final Eth1Provider eth1Provider;
+  private Optional<UInt64> headAtFollowDistance = Optional.empty();
+  private final AtomicBoolean reachedHead = new AtomicBoolean(false);
+
+  private final Subscribers<HeadUpdatedSubscriber> subscribers = Subscribers.create(true);
+
+  public BlockBasedEth1HeadTracker(final AsyncRunner asyncRunner, final Eth1Provider eth1Provider) {
+    this.asyncRunner = asyncRunner;
+    this.eth1Provider = eth1Provider;
+  }
+
+  @Override
+  public void start() {
+    if (!running.compareAndSet(false, true)) {
+      return;
+    }
+    pollLatestHead();
+  }
+
+  private void pollLatestHead() {
+    if (!running.get()) {
+      return;
+    }
+    eth1Provider
+        .getLatestEth1Block()
+        .thenAccept(this::onLatestBlockHead)
+        .exceptionally(
+            error -> {
+              LOG.debug("Failed to get latest Eth1 chain head. Will retry.", error);
+              return null;
+            })
+        .always(
+            () ->
+                asyncRunner
+                    .runAfterDelay(
+                        this::pollLatestHead,
+                        Duration.ofSeconds(Constants.SECONDS_PER_ETH1_BLOCK.longValue()))
+                    .finish(
+                        () -> {},
+                        error ->
+                            LOG.error("Scheduling next check of Eth1 chain head failed", error)));
```

Comment:
This might become a little flood-like if it happens, it might be something that needs to be only actually logged periodically?

Reply:
If it happens we're unable to execute new scheduled tasks so can't retry and everything is going to go pretty badly.  Suspect it should actually be fatal level as eth1 tracking definitely won't recover.

Reply:
will be worth seeing if it happens i guess, can leave it for now, just a pattern that sometimes leads to us needing to filter